lib/spack/spack/build_systems/cuda.py

```diff
@@ -0,0 +1,81 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack.package import Package
+from spack.directives import depends_on, variant, conflicts
+
+
+class CudaPackage(Package):
+    """Auxiliary class which contains CUDA variant, dependencies and conflicts
+    and is meant to unify and facilitate its usage.
+    """
+
+    # FIXME: keep cuda and cuda_arch separate to make usage easier untill
+    # Spack has depends_on(cuda, when='cuda_arch!=None') or alike
+    variant('cuda', default=False,
+            description='Build with CUDA')
+    # see http://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#gpu-feature-list
+    variant('cuda_arch', default=None,
+            description='CUDA architecture',
+            values=('20', '30', '32', '35', '50', '52', '53', '60', '61',
+                    '62', '70'),
+            multi=True)
+
+    # see http://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#nvcc-examples
+    # and http://llvm.org/docs/CompileCudaWithLLVM.html#compiling-cuda-code
+    @property
+    def cuda_flags(self, arch_list):
+        return ['--generate-code arch=compute_{0},code=sm_{0} '
+                '--generate-code arch=compute_{0},code=compute_{0}'.format(s)
```

Comment:
careful, this will not replace the first line!
add brackets around both strings :)

Reply:
thanks